x-pack/plugins/observability_solution/infra/server/lib/alerting/inventory_metric_threshold/inventory_metric_threshold_executor.ts

```diff
@@ -96,6 +107,13 @@ export const createInventoryMetricThresholdExecutor =
       getTimeRange,
     } = options;
 
+    const { share } = libs.plugins;
+    const alertsLocator = share.setup.url.locators.get<AlertsLocatorParams>(alertsLocatorID);
```

Comment:
nit: maybe we can pass all the locators in the lib.locators, so we avoid getting this locator like this in different places. (I see the same implementation in metric_threshold_executor.ts)

So, it will be something like:
```
libs.locators: {
   alerts,
   assetDetails,
   inventory
}
```

Reply:
I'd rather leave it up to the downstream service to decide what to do with the plugins and call the locators as they see fit. Currently, every time there is a new plugin or a new locator we need to change the `libs` type, I want to avoid that as much as possible.

Reply:
Hmm, is it only to avoid updating types? I don't think we will have many locators and usually, I try to have limited and more clear dependencies and avoid code duplication.

Maybe an idea would be to have a function to extract locators, but up to you!